db/data_migration/20150730134022_republish_routes_for_all_detailed_guides.rb

```diff
@@ -0,0 +1,21 @@
+require 'plek'
+require 'gds_api/router'
+
+router_api = GdsApi::Router.new(Plek.find('router-api'))
+
+scope = Document.distinct.
+  joins(:editions).
+  where(document_type: "DetailedGuide",
+    editions: {state: Edition::POST_PUBLICATION_STATES})
+
+count = scope.count
+
+scope.each_with_index do |guide, i|
+  if guide.ever_published_editions.any?
+    slug = guide.slug.sub(%r{^deleted-}, '')
+    puts "Adding route #{i+1}/#{count} for /#{slug}"
+    router_api.add_route("/#{slug}", "exact", "whitehall-frontend")
```

Comment:
Actually, my comment above isn't quite correct.  These calls need to go via either publishing-api or panopticon (both of which check with url-arbiter). I believe detailed guides currently get registered in panopticon, so that may be the easier route, however note that panopticon won't register routes for detailed guides that are set to archived.

An alternative approach here would be to continue to register directly with the router, but to check directly with url-arbiter before doing so.  Grab me at some point today if you want to go through the options here.